force-app/main/default/classes/DeceasedBatch_TEST.cls

```diff
@@ -0,0 +1,238 @@
+    }
+
+    @IsTest
+    private static void shouldUpdateAllMembersDeceasedWhenMembersAreMixed() {
+        List<Contact> mixedContacts1 = getContactsWithAccount(3);
+        mixedContacts1[0].Account.All_Members_Deceased__c = true;
+        List<Contact> mixedContacts2 = getContactsWithAccount(3);
+        mixedContacts2[0].Account.All_Members_Deceased__c = true;
+
+        List<Contact> allContacts = mixedContacts1.clone();
+        allContacts.addAll(mixedContacts2);
+        setContactsDeceased(allContacts, false, true);
+
+        DeceasedBatch batch = new DeceasedBatch();
+        Map<Id, Boolean> deceasedById = batch.getAllMembersDeceasedById(allContacts);
+        System.assertEquals(2, deceasedById.size(), 'Two Account entries should exist in the Id => Boolean map');
+
+        List<Account> accountUpdates = batch.getAllMembersDeceasedUpdates(allContacts, deceasedById);
+        System.assertEquals(2, accountUpdates.size(), 'Two Account updates should be identified');
+        System.assertEquals(false, accountUpdates[0].All_Members_Deceased__c, 'All Members Deceased should be false');
+        System.assertEquals(false, accountUpdates[1].All_Members_Deceased__c, 'All Members Deceased should be false');
+    }
+
+    @IsTest
+    private static void shouldNotUpdateAllMembersDeceasedWhenMembersAreMixed() {
+        List<Contact> mixedContacts1 = getContactsWithAccount(3);
+        List<Contact> mixedContacts2 = getContactsWithAccount(3);
+
+        List<Contact> allContacts = mixedContacts1.clone();
+        allContacts.addAll(mixedContacts2);
+        setContactsDeceased(allContacts, false, true);
+
+        DeceasedBatch batch = new DeceasedBatch();
+        Map<Id, Boolean> deceasedById = batch.getAllMembersDeceasedById(allContacts);
+        System.assertEquals(2, deceasedById.size(), 'Two Account entries should exist in the Id => Boolean map');
+
+        List<Account> accountUpdates = batch.getAllMembersDeceasedUpdates(allContacts, deceasedById);
+        System.assertEquals(0, accountUpdates.size(), 'No Account updates should be identified');
+    }
+
+    ///// Helpers
+
+    private static List<Contact> getContactsWithAccount(Integer contactsPerAccount) {
+        Account testAccount = UTIL_UnitTestData_TEST.buildHouseholdAccount();
+        testAccount.All_Members_Deceased__c = false;
+        testAccount.Id = UTIL_UnitTestData_TEST.mockId(Account.SObjectType);
+
+        List<Contact> contacts = UTIL_UnitTestData_TEST.getContacts(contactsPerAccount);
+        for (Contact testContact : contacts) {
+            testContact.Id = UTIL_UnitTestData_TEST.mockId(Contact.SObjectType);
+            testContact.Account = testAccount;
+            testContact.AccountId = testAccount.Id;
+            testContact.Deceased__c = false;
+        }
+
+        return contacts;
+    }
+
+    private static void setContactsDeceased(List<Contact> contacts, Boolean accountMatch, Boolean mixed) {
```

Comment:
This could use a description comment, walking through it I get what you're doing, but it's a little confusing!
Alternatively, you could setup a few abstracted helpers that set the Booleans, `setAllContactsDeceased`,  `setSomeContactsDeceased` and `matchContactsDeceasedToHouseholds` or something like that.